### ISSUE TITLE
Failure getting list of active applications

### ISSUE THREAD

**Issue by AvrahamAtia** [facebook/WebDriverAgent#1054]
The problem
Failure getting list of active applications

Environment
Appium server: 1.9.1
Node: v10.4.1
Allure: 2.8.0
xcode-select: 10.1
Apache Maven 3.5.3 (3383c37e1f9e9b3bc3df5050c29c8aff9f295297; 2018-02-24T21:49:05+02:00)
Maven home: /usr/local/Cellar/maven/3.5.3/libexec
Java version: 11.0.1, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk-11.0.1.jdk/Contents/Home
Default locale: en_IL, platform encoding: UTF-8
OS name: "mac os x", version: "10.13.6", arch: "x86_64", family: "mac"

Details
Parallel execution iOS real devices and simulator.

Link to Appium logs
[Xcode] t = 67104.91s Open com.shutterfly.mopho.prototype
[Xcode] t = 67104.91s Launch com.shutterfly.mopho.prototype
[Xcode]
[Xcode] t = 67104.99s Wait for accessibility to load
[Xcode]
[Xcode] t = 67106.41s Wait for com.shutterfly.mopho.prototype to idle
[Xcode]
[Xcode] t = 67166.42s App event loop idle notification not received, will attempt to continue.
[Xcode]
[Xcode] t = 67166.42s App animations complete notification not received, will attempt to continue.
[Xcode] 2019-01-01 11:32:16.204292+0200 WebDriverAgentRunner-Runner[23355:193499] Enqueue Failure: Failure to determine system application: (null) /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 37 1
[Xcode]
[Xcode] 2019-01-01 11:33:21.207391+0200 WebDriverAgentRunner-Runner[23355:193499] Enqueue Failure: Failure fetching attributes for (null): Error Domain=com.apple.XCTestErrorDomain Code=1000 "Timeout waiting for fulfillment of promise for 'Fetching attributes for (null)'." UserInfo={NSLocalizedDescription=Timeout waiting for fulfillment of promise for 'Fetching attributes for (null)'.} /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 37 1
[Xcode]
[Xcode] 2019-01-01 11:33:21.208218+0200 WebDriverAgentRunner-Runner[23355:193499] Enqueue Failure: Failure getting list of active applications: (null) /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 37 1
[Xcode]

**Comment by shibupanda** [comment]
@AvrahamAtia Even I am getting the same "Enqueue Failure:" while executing my Appium script and it took much time to launch. Please let us know if the issue you have resolved.

**Comment by srinidhipatel** [comment]
We are also facing same error.

[Xcode] 2019-03-28 13:34:07.473372+0530 WebDriverAgentRunner-Runner[40852:3532381] Enqueue Failure: Failure getting list of active applications: (null) /usr/local/Cellar/node@8/8.15.0/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 37 1
We are blocked on this..Can you please help on this..

**Comment by AvrahamAtia** [comment]
I do not have solution for that, For now I testing only on Real devices.

**Comment by jerolan** [comment]
I'm having the same problem, does anyone have a solution?

**Comment by AvrahamAtia** [comment]
@jerolan Which version of simulators are you using?

**Comment by jerolan** [comment]
@AvrahamAtia  I'm on iOS 12.1

**Comment by AvrahamAtia** [comment]
Try not use 12.1 version, use 11.3.1 or 11.3.

**Comment by jerolan** [comment]
No, I still have the same problem, am I using react-native can it be related to that?

**Comment by jade685** [comment]
Having same problem here, can run though, just really slow.

**Comment by Srinija25** [comment]
> No, I still have the same problem, am I using react-native can it be related to that?

I am also facing the same issue, Working on React Native app, iOS version 12.1, iPhone 6 and it is taking almost 20 mins to start the execution, Please let me know if you have any solution.

**Comment by shibupanda** [comment]
Could you please upgrade your Appium to latest(1.12.0). As XCUITest driver has given a capability called "eventLoopIdleDelaySec"
Please refer below link.
https://github.com/appium/appium/releases

**Comment by Srinija25** [comment]
Thanks for the reply, @shibupanda I am using the latest Appium version 1.12.0

**Comment by shibupanda** [comment]
@Srinija25 Have you used "eventLoopIdleDelaySec" capability.

**Comment by Srinija25** [comment]
yes I have used it my capabilities are:

capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "iphone7-test");
				capabilities.setCapability(MobileCapabilityType.PLATFORM_NAME, "IOS");
				capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, "12.1.4");
				capabilities.setCapability(MobileCapabilityType.APP, fs.getAbsolutePath());
				capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, AutomationName.IOS_XCUI_TEST);
				capabilities.setCapability("deviceOrientation", "portrait");
				capabilities.setCapability("xcodeOrgId", "*****");
				capabilities.setCapability("xcodeSigningId", "*****");
				
				capabilities.setCapability("udid", "");
				capabilities.setCapability("updateWDABundleId", "********");
				capabilities.setCapability("waitForQuiescence", "false");
				capabilities.setCapability("useNewWDA", "true");
				capabilities.setCapability("wdaEventloopIdleDelay", "3");
				capabilities.setCapability("eventLoopIdleDelaySec", "1");
				capabilities.setCapability("useJSONSource", "true");

**Comment by kamaleshrnp** [comment]
**Environment Details :**

MacOs Mojave 10.14.4
Appium Server 1.12.1
Xcode 10.2
Appium Java Client 6.1.0

**Capabilities:**

caps.setCapability("automationName", prop.getProperty("ios.automationName"));
caps.setCapability("platformVersion", prop.getProperty("ios.platformVersion"));
caps.setCapability("platformName", prop.getProperty("ios"));
caps.setCapability("deviceName", prop.getProperty("ios.deviceName"));
caps.setCapability("autoAcceptAlerts", prop.getProperty("ios.autoAcceptAlerts"));
caps.setCapability("showXcodeLog","true");
caps.setCapability("useNewWDA","true");
caps.setCapability("waitForQuietness","false");
caps.setCapability("wdaEventloopIdleDelay","3");
caps.setCapability("eventLoopIdleDelaySec", "1");
caps.setCapability("noReset", prop.getProperty("ios.noReset"));

**Comments**

1) Manually, It works perfectly in appium desktop version 1.12.1
2) I tried with various Xcode Versions ; but nothing helped(got errors like Couldn't identify simulators with particular versions. But they are actually installed)
3) Tried removing capabilities "useNewWDA" and reduce the delay for capabilities("wdaEventloopIdleDelay & eventLoopIdleDelay)
Need Help
